src/main/java/com/adyen/service/Checkout.java

```diff
@@ -0,0 +1,132 @@
+
+import java.io.IOException;
+
+public class Checkout extends ApiKeyAuthenticatedService {
+
+    private Payments payments;
+    private PaymentMethods paymentMethods;
+    private PaymentsDetails paymentsDetails;
+    private PaymentSession paymentSession;
+    private PaymentsResult paymentsResult;
+
+    public Checkout(Client client) {
+
+        super(client);
+        payments = new Payments(this);
+        paymentMethods = new PaymentMethods(this);
+        paymentsDetails = new PaymentsDetails(this);
+        paymentSession = new PaymentSession(this);
+        paymentsResult = new PaymentsResult(this);
+    }
+
+    /**
+     * POST /payments API call
+     *
+     * @param paymentsRequest
+     * @return paymentResponse
+     * @throws IOException
+     * @throws ApiException
+     */
+    public PaymentsResponse payments(PaymentsRequest paymentsRequest) throws ApiException, IOException {
+        String jsonRequest = GSON.toJson(paymentsRequest);
+        String jsonResult = payments.request(jsonRequest);
+        return GSON.fromJson(jsonResult, new TypeToken<PaymentsResponse>() {
+        }.getType());
+
+    }
+
+    /**
+     * POST /paymentMethods API call
+     *
+     * @param paymentMethodsRequest
+     * @return paymentMethodsResponse
+     * @throws IOException
+     * @throws ApiException
+     */
+
+    public PaymentMethodsResponse paymentMethods(PaymentMethodsRequest paymentMethodsRequest) throws ApiException, IOException {
+        String jsonRequest = GSON.toJson(paymentMethodsRequest);
+        String jsonResult = paymentMethods.request(jsonRequest);
+        return GSON.fromJson(jsonResult, new TypeToken<PaymentMethodsResponse>() {
+        }.getType());
+
+    }
+
+    /**
+     * POST payments/details API call
+     *
+     * @param paymentsDetailsRequest
+     * @return paymentResponse
+     * @throws IOException
+     * @throws ApiException
+     */
+    public PaymentsResponse paymentsDetails(PaymentsDetailsRequest paymentsDetailsRequest) throws ApiException, IOException {
+        String jsonRequest = GSON.toJson(paymentsDetailsRequest);
+        String jsonResult = paymentsDetails.request(jsonRequest);
+        return GSON.fromJson(jsonResult, new TypeToken<PaymentsResponse>() {
+        }.getType());
+
+    }
+
+    /**
+     * POST /paymentSession API call
+     *
+     * @param paymentSessionRequest
+     * @return paymentSetupResponse
+     * @throws IOException
+     * @throws ApiException
+     */
+
+    public PaymentSessionResponse paymentSession(PaymentSessionRequest paymentSessionRequest) throws ApiException, IOException {
+        String jsonRequest = GSON.toJson(paymentSessionRequest);
+        String jsonResult = paymentSession.request(jsonRequest);
+        return GSON.fromJson(jsonResult, new TypeToken<PaymentSessionResponse>() {
+        }.getType());
+    }
+
+    /**
+     * POST payments/result API call
+     *
+     * @param paymentResultRequest
+     * @return paymentVerificationResponse
```

Comment:
we better rename this doc-block return to PaymentResultResponse or just remove the param,return,throws since they don't add any additional info

Reply:
done